### DISCUSSION TITLE
MySQL usage example

### DISCUSSION THREAD

**Discussion by SkyJust4Fun** [ianlucas/cs2-inventory-simulator#92] [General]
My way to make it work with mysql on 1.0.0:

1. Change schema.prisma

 12 - provider = "postgres" -> provider = "mysql"

 24 - 'inventory String' -> 'inventory String? @db.LongText'
 
 46 - 'body String' -> 'body String @db.LongText'
 
 In mysql String will translate to varchar which is 191 lenght, so String @db.LongText translates more correctly to 4GB of text compatible with string of postgres of 1GB of text.
 
 There is no need to change more types because there wont be more then 191 characters on any of it if needed change to String @db.LongText
 
 2. Delete migrations folder in prisma/
 3. npx prisma migrate dev (Build new migrations to mysql) (ENV File must be updated with correct info atm)
 4. Continue normal installation from (npx prisma migrate deploy)
 
 
 //Full changes
 
 model User {
  authTokens ApiAuthToken[]
  avatar String
  caches UserCache[]
  createdAt DateTime @default(now())
  groups UserGroup[]
  id String @id
  inventories UserDomainInventory[]
  inventory String
  name String
  overwrites UserRule[]
  preferences UserPreference?
  syncedAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

To:

model User {
  authTokens ApiAuthToken[]
  avatar String
  caches UserCache[]
  createdAt DateTime @default(now())
  groups UserGroup[]
  id String @id
  inventories UserDomainInventory[]
  inventory String? @db.LongText
  name String
  overwrites UserRule[]
  preferences UserPreference?
  syncedAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model UserCache {
  @@id([url, userId])
  args String?
  body String
  timestamp DateTime
  url String
  user User @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId String 
}

To:

model UserCache {
  @@id([url, userId])
  args String?
  body String @db.LongText 
  timestamp DateTime
  url String
  user User @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId String
}

**Comment by SkyJust4Fun** [comment]
Im having a issue now let me check first ..

**Comment by ianlucas** [comment]
I'm planning to add a branch for MySQL migrations soon (I'll try to work on it Thursday).

**Comment by SkyJust4Fun** [comment]
Sure that one above is working as intended but im no longer able to login :v not even on the old one and the 1.0.0

**Reply by SkyJust4Fun** [reply]
Thats why i said i found a issue..

**Comment by SkyJust4Fun** [comment]
@ianlucas There is a mistake above preventing from login in, on:

24 - 'inventory String' -> 'inventory String @db.LongText' should be 'inventory String? @db.LongText'

**Reply by SkyJust4Fun** [reply]
Without the ? mysql wont accept it being null, from that point creating an error logging in.

**Reply by ianlucas** [reply]
You can edit the original post.

**Reply by SkyJust4Fun** [reply]
Done xD
